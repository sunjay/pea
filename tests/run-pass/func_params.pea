fn foo() {
    println!(b"foo");
}

fn foo2(a, b) {
    println!(b"foo2");
    println!(a);
    println!(b);

    // Function parameters are still regular variables
    a = b"cool beans";
    // Function parameters can be shadowed by variables
    let b = b"shadowed b";
    println!(a);
    println!(b);
}

fn foo3(
    a,
    b,
    c,
) {
    println!(b"foo3");
    println!(a);
    println!(b);
    println!(c);
}

fn main() {
    let x = b"wooooo";
    foo2(3, x);
    foo();
    foo3(x, x, 4);
}
