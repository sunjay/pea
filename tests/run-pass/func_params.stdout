foo2
3
wooooo
cool beans
shadowed b
foo
foo3
wooooo
wooooo
4
